### PR TITLE
Add newt docs command to automatically build using sphinx

### PR DESCRIPTION
**Pull request by sterlinghughes**
This automatically builds documents within a mynewt repo, including any sub-repos using the sphinx format.

### CODE REVIEW

**Comment by utzig** [comment]
Whata is your main design idea, to build docs for local consumption or to push to the main docs website?

File: util/util.go [hunk 1]
```diff
@@ -502,6 +502,21 @@ func MoveDir(srcDir string, destDir string) error {
 	return nil
 }
 
+func CallInDir(path string, execFunc func() error) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	os.Chdir(path)
```
Comment (ccollins476ad) [inline util/util.go]: It is probably a good idea to check for errors here.

Also, this is just a stylistic comment - I think it is more "Go like" to switch back to the original directory using `defer`.  Then you can continue to return on error.

File: newt/docs/docsbuilder.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+func (db *DocsBuilder) GetDocs() ([]*DocDescriptor, error) {
+	docs := []*DocDescriptor{}
+
+	for _, repo := range db.project.Repos() {
+		name := repo.Name()
+
+		if repo.IsLocal() {
+			name = db.project.Name()
+		}
+		fmt.Println(name)
```
Comment (utzig) [inline newt/docs/docsbuilder.go]: Should be a debug message

**Comment by utzig** [comment]
@ccollins476ad I think this is a good start, are you OK with merging and updating later?

**Comment by ccollins476ad** [comment]
@utzig, sounds good.

**Pull request merged**